src/onboarding.ts

```diff
@@ -228,14 +229,66 @@ promptfoo eval
 Afterwards, you can view the results by running \`promptfoo view\`
 `;
 
-function recordOnboardingStep(step: string, properties: Record<string, EventValue> = {}) {
+function recordOnboardingStep(step: string, properties: any = {}) {
   telemetry.recordAndSend('funnel', {
     type: 'eval onboarding',
     step,
     ...properties,
   });
 }
 
+/**
+ * Iterate through user choices and determine if the user has selected a provider that needs an API key
+ * but has not set and API key in their enviorment.
+ */
+export function reportProviderAPIKeyWarnings(providerChoices: (string | object)[]): string[] {
+  // Dictionary of what warnings may have already been printed for the user.
+  const shownWarnings = new Map([
+    [
+      ProvidersRequiringAPIKeysEnvKeys.OPENAI_API_KEY as keyof typeof ProvidersRequiringAPIKeysEnvKeys,
```

Comment:
why does this need to be typed?

Reply:
This is a weird artifact of typescript enums and attempting to use them as strings -- essentially this enables us "to get a Type that represents all Enum keys as strings."1

I like the ability to use enums although the syntax is a bit clunky as they don't have the ability to modified like a regular map or dictionary. Hopefully this syntax might get better in the future.

1 https://www.typescriptlang.org/docs/handbook/enums.html#enums-at-compile-time